yt_idv/scene_components/blocks.py

```diff
@@ -34,34 +34,58 @@ class BlockRendering(SceneComponent):
 
     def render_gui(self, imgui, renderer, scene):
         changed = super().render_gui(imgui, renderer, scene)
+
+        max_sf = 50.0 if self._yt_geom_str == "spherical" else 20.0
         _, sample_factor = imgui.slider_float(
-            "Sample Factor", self.sample_factor, 1.0, 20.0
+            "Sample Factor", self.sample_factor, 1.0, max_sf
         )
         if _:
             self.sample_factor = sample_factor
         # Now, shaders
-        shader_combos = list(sorted(component_shaders[self.name]))
+        valid_shaders = get_shader_combos(
+            self.name, coord_system=self.data._yt_geom_str
+        )
         descriptions = [
-            component_shaders[self.name][_]["description"] for _ in shader_combos
+            component_shaders[self.name][_]["description"] for _ in valid_shaders
         ]
-        selected = shader_combos.index(self.render_method)
+        selected = valid_shaders.index(self.render_method)
         _, shader_ind = imgui.listbox("Shader", selected, descriptions)
         if _:
-            self.render_method = shader_combos[shader_ind]
+            self.render_method = valid_shaders[shader_ind]
         changed = changed or _
         if imgui.button("Add Block Outline"):
-            from ..scene_annotations.block_outline import BlockOutline
+            if self.data._yt_geom_str == "cartesian":
+                from ..scene_annotations.block_outline import BlockOutline
+
+                block_outline = BlockOutline(data=self.data)
+                scene.annotations.append(block_outline)
+            elif self.data._yt_geom_str == "spherical":
+                from ..scene_data.block_collection import _block_collection_outlines
 
-            block_outline = BlockOutline(data=self.data)
-            scene.annotations.append(block_outline)
+                cc, cc_render = _block_collection_outlines(
+                    self.data, outline_type="blocks"
+                )
+                scene.data_objects.append(cc)
+                scene.components.append(cc_render)
+
+        # the following only work for cartesian data at present
```

Comment:
Is this still true? :smile:

Reply:
oh! no! the last few commits made these work :)